R/model.R

```diff
@@ -1149,6 +1149,8 @@ sample <- function(data = NULL,
                    show_messages = TRUE,
                    show_exceptions = TRUE,
                    diagnostics = c("divergences", "treedepth", "ebfmi"),
+                   save_metric = TRUE,
+                   save_cmdstan_config = TRUE,
```

Comment:
```suggestion
                   save_metric = if (cmdstan_version() > "2.34.0") TRUE else NULL,
                   save_cmdstan_config = if (cmdstan_version() > "2.34.0") TRUE else NULL,
```

Then the function arguments below would just use the `save_metric` and `save_cmdstan_config` values directly

Reply:
And the same for the other methods as well (laplace, optimize, etc.)

Reply:
good catch, missed that when I was looking at it